src/Spatial.Tests/Euclidean/Circle2DTests.cs

```diff
@@ -58,5 +59,51 @@ public void CircleFromThreePointsArgumentException()
 
             Assert.Throws<ArgumentException>(() => { Circle2D.FromPoints(p1, p2, p3); });
         }
+
+        [TestCase("0,0", 1, "-10,+10", "+10,+10", 0)]
+        [TestCase("0,0", 1, "-10,+1", "+10,+1", 1)]
+        [TestCase("0,0", 1, "-10,0", "+10,0", 2)]
+        [TestCase("0,0", 1, "-10,-1", "+10,-1", 1 )]
+        [TestCase("0,0", 1, "-10,-10", "+10,-10", 0)]
+        public void CircleIntersectWithLine2D_NumberOfIntersections(string sc, double radius, string sps, string spe, int expectedNumberOfIntersections)
+        {
+            var circle = new Circle2D(Point2D.Parse(sc), radius);
+            var line = new Line2D(Point2D.Parse(sps), Point2D.Parse(spe));
+
+            var actual = circle.IntersectWith(line);
+
+            Assert.That(actual.Count(), Is.EqualTo(expectedNumberOfIntersections));
+            //TODO: the intersection should be on the circle
+            Assert.That(actual.All(p => Math.Abs(circle.Center.DistanceTo(p) - circle.Radius) < 1e-6), Is.EqualTo(true), "distance between center and intersection");
+            //TODO: the intersection should be on the line
+        }
+
+        //segment contains the all intersections(same to the cases of circle and line)
+        [TestCase("0,0", 1, "-10,+10", "+10,+10", 0)]
+        [TestCase("0,0", 1, "-10,+1", "+10,+1", 1)]
+        [TestCase("0,0", 1, "-10,0", "+10,0", 2)]
+        [TestCase("0,0", 1, "-10,-1", "+10,-1", 1)]
+        [TestCase("0,0", 1, "-10,-10", "+10,-10", 0)]
+        //segments cross the circle's contour just 1 time
+        [TestCase("0,0", 1, "+0,+10", "+10,+10", 0)]
+        [TestCase("0,0", 1, "+0,+1", "+10,+1", 1)]
+        [TestCase("0,0", 1, "+0,0", "+10,0", 1)]
+        [TestCase("0,0", 1, "+0,-1", "+10,-1", 1)]
+        [TestCase("0,0", 1, "+0,-10", "+10,-10", 0)]
+        //segment contains no intersections(px of the startingPoint is too big to intersect with the circle)
+        [TestCase("0,0", 1, "+10,+10", "+100,+10", 0)]
+        [TestCase("0,0", 1, "+10,+01", "+100,+1", 0)]
```

Comment:
+01?

Reply:
Which part of "+01" di you point out, the center "+10,+01" or the right "+100,+1"?

Reply:
done in the commit https://github.com/f-frhs/mathnet-spatial/commit/f3815edfaac3955fbd2f08a14fd2280aac9d84bc.

I'm not sure that I understood completely what you mean.
If you find any flaw, please let me know.